stripzip.py

```diff
@@ -10,34 +10,104 @@ class NonZipFileError(ValueError):
 
 
 def _zero_zip_date_time(zip_):
-    archive_size = os.fstat(zip_.fileno()).st_size
+    def purify_extra_data(mm, offset, length):
+        extra_header_struct = Struct("<HH")
+        # 0. id
+        # 1. length
+        STRIPZIP_OPTION_HEADER = 0xFFFF
+        EXTENDED_TIME_DATA = 0x5455
+        # Some sort of extended time data, see
+        # ftp://ftp.info-zip.org/pub/infozip/src/zip30.zip ./proginfo/extrafld.txt
+        # fallthrough
+        UNIX_EXTRA_DATA = 0x7875
+        # Unix extra data; UID / GID stuff, see
+        # ftp://ftp.info-zip.org/pub/infozip/src/zip30.zip ./proginfo/extrafld.txt
+        mlen = offset + length
+
+        while offset < mlen:
+            values = list(extra_header_struct.unpack_from(mm, offset))
+            _, header_length = values
+            extra_struct = Struct("<HH" + "B"*header_length)
```

Comment:
See previous comment, I'm using the exact same strategy of native stripzip, so both they will output a similar stripped zip.

To be noted that maximum header length is 2^16-2, so 8kB-$(header size [idx+length]), so even if huge it's an 8kB extra data, not really something worrying me. Crafted zips may issue larger memory read, but I don't think is really an issue.

About using `mmap.move` or `mmap.resize` I'm really not sure if it will work, and I prefer to maintain same behaviour of C implementation if possible.